cpp/src/arrow/compute/exec.cc

```diff
@@ -118,8 +133,15 @@ std::string ExecBatch::ToString() const {
 ExecBatch ExecBatch::Slice(int64_t offset, int64_t length) const {
   ExecBatch out = *this;
   for (auto& value : out.values) {
-    if (value.is_scalar()) continue;
-    value = value.array()->Slice(offset, length);
+    if (value.is_scalar()) {
+      continue;
```

Comment:
Should we replace this with `// Do nothing` or something?

Reply:
You mean remove the `continue` statement? I suppose we could.